src/main/scala/com/scality/clueso/MetadataIngestionPipeline.scala

```diff
@@ -0,0 +1,187 @@
+package com.scality.clueso
+
+import com.fasterxml.jackson.databind.node.{ArrayNode, NullNode, ObjectNode, TextNode}
+import com.fasterxml.jackson.databind.{JsonNode, ObjectMapper}
+import com.typesafe.config.Config
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.hadoop.fs.Path
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.streaming.{OutputMode, ProcessingTime}
+import org.apache.spark.sql.types.TimestampType
+import org.apache.spark.sql.{DataFrame, SparkSession}
+
+import scala.collection.mutable
+
+class EventMessageRewriter {
+  val mapper = new ObjectMapper()
+
+  def rewriteMsg(bytes: Array[Byte]): String = {
+    val payload = new String(bytes)
+
+    var rootNode = mapper.readTree(payload).asInstanceOf[ObjectNode]
+
+    val valueTxtNode = rootNode.path("value").asInstanceOf[TextNode]
+
+    val valueNode = mapper.readTree(valueTxtNode.asText()).asInstanceOf[ObjectNode]
+
+    val metadataFieldNames = mutable.ListBuffer[String]()
+
+    val it = valueNode.fieldNames()
+    while (it.hasNext) {
+      val fieldName = it.next()
+      if (fieldName.startsWith("x-amz-meta-")) {
+        metadataFieldNames += fieldName
+      } else if (fieldName.equals("location")) {
+        val locationArray = valueNode.path("location")
+
+        if (!locationArray.isInstanceOf[NullNode] && locationArray.asInstanceOf[ArrayNode].size() > 1) {
+          val _locationArray =  locationArray.asInstanceOf[ArrayNode]
+          (1 until locationArray.size()).foreach(i => _locationArray.remove(i))
+          valueNode.replace("location", _locationArray)
+        }
+      }
+    }
+
+    val userMd = if (rootNode.has("userMd")) {
```

Comment:
this would be if our metadata actually had this attribute? we shouldn't but fine to keep here.

Reply:
It was a conditional created when I wasn't sure what came in the stream – we can remove it now backbeat is not building this

Reply:
done